mods/custom-desktop-watermark.wh.cpp

```diff
@@ -0,0 +1,199 @@
+// ==WindhawkMod==
+// @id              custom-desktop-watermark
+// @name            Custom Desktop Watermark
+// @description     Lets you set your own desktop watermark text
+// @version         1.0.0
+// @author          aubymori
+// @github          https://github.com/aubymori
+// @include         explorer.exe
+// @compilerOptions -lgdi32 -DWINVER=0x0A00
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Custom Desktop Watermark
+This mod allows you to completely customize the desktop watermark,
+putting in your own lines of text.
+
+## Examples
+
+**Windows 7 (not genuine)**:
+![Windows 7 (not genuine)](https://raw.githubusercontent.com/aubymori/images/main/custom-desktop-watermark-win7.png)
+
+**Windows XP Professional x64 Edition**:
+![Windows XP Professional x64 Edition](https://raw.githubusercontent.com/aubymori/images/main/custom-desktop-watermark-winxp.png)
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- lines:
+  - - text: ""
+      $name: Text
+    - title: false
+      $name: Use caption font
+      $description: Use caption font instead of message font when classic fonts are enabled.
+  $name: Text lines
+- classic: false
+  $name: Use classic fonts
+  $description: Use caption/message font instead of just caption font.
+*/
+// ==/WindhawkModSettings==
+
+#include <windhawk_utils.h>
+#include <windows.h>
+#include <vector>
+
+#define RECTWIDTH(rc)   ((rc).right - (rc).left)
+#define RECTHEIGHT(rc)  ((rc).bottom - (rc).top)
+
+WINUSERAPI BOOL WINAPI SystemParametersInfoForDpi(UINT uiAction, UINT uiParam, PVOID pvParam, UINT fWinIni, UINT dpi);
+
+struct WatermarkLine
+{
+    std::wstring text;
+    bool title;
+};
+
+std::vector<WatermarkLine> g_lines;
+bool g_bClassic = false;
+
+bool (*CDesktopWatermark_s_WantWatermark_orig)(void);
+bool CDesktopWatermark_s_WantWatermark_hook(void)
```

Comment:
Calling conventions are missing, which is likely to cause a crash for 32-bit targets. Please add calling conventions or, if the mod doesn't support 32-bit, limit it to 64-bit, which will also make compilation faster. See:
https://github.com/ramensoftware/windhawk/wiki/Development-tips#hooks-and-calling-conventions